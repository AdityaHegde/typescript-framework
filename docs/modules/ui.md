[@adityahegde/typescript-framework](../README.md) / [Modules](../modules.md) / ui

# Module: ui

## Table of contents

### Classes

- [ComponentByModelMap](../classes/ui.ComponentByModelMap.md)
- [LightErrorBoundary](../classes/ui.LightErrorBoundary.md)
- [SelectEnumItems](../classes/ui.SelectEnumItems.md)

### Functions

- [DataPage](ui.md#datapage)
- [ModalWrapper](ui.md#modalwrapper)
- [fetchGet](ui.md#fetchget)
- [fetchPost](ui.md#fetchpost)
- [fetchWrapper](ui.md#fetchwrapper)
- [getDisplayText](ui.md#getdisplaytext)
- [getQueryString](ui.md#getquerystring)
- [useRoutePath](ui.md#useroutepath)

## Functions

### DataPage

▸ **DataPage**<`Model`\>(`__namedParameters`): `Element`

#### Type parameters

| Name | Type |
| :------ | :------ |
| `Model` | extends [`BaseType`](../classes/models.BaseType.md)<`Model`\> |

#### Parameters

| Name | Type |
| :------ | :------ |
| `__namedParameters` | `Object` |
| `__namedParameters.modelStore` | [`ModelStore`](../classes/ui_store.ModelStore.md)<`Model`\> |
| `__namedParameters.params?` | `any` |

#### Returns

`Element`

#### Defined in

[src/ui/DataPage.tsx:11](https://github.com/AdityaHegde/typescript-framework/blob/3b13972/src/ui/DataPage.tsx#L11)

___

### ModalWrapper

▸ **ModalWrapper**<`Model`\>(`__namedParameters`): `Element`

#### Type parameters

| Name | Type |
| :------ | :------ |
| `Model` | extends [`BaseType`](../classes/models.BaseType.md)<`Model`\> |

#### Parameters

| Name | Type |
| :------ | :------ |
| `__namedParameters` | `ModalWrapperProps`<`Model`\> |

#### Returns

`Element`

#### Defined in

[src/ui/ModalWrapper.tsx:17](https://github.com/AdityaHegde/typescript-framework/blob/3b13972/src/ui/ModalWrapper.tsx#L17)

___

### fetchGet

▸ **fetchGet**(`url`): `Promise`<`any`\>

#### Parameters

| Name | Type |
| :------ | :------ |
| `url` | `any` |

#### Returns

`Promise`<`any`\>

#### Defined in

[src/ui/utils/fetch.ts:1](https://github.com/AdityaHegde/typescript-framework/blob/3b13972/src/ui/utils/fetch.ts#L1)

___

### fetchPost

▸ **fetchPost**(`url`, `bodyJson`): `Promise`<`any`\>

#### Parameters

| Name | Type |
| :------ | :------ |
| `url` | `any` |
| `bodyJson` | `any` |

#### Returns

`Promise`<`any`\>

#### Defined in

[src/ui/utils/fetch.ts:5](https://github.com/AdityaHegde/typescript-framework/blob/3b13972/src/ui/utils/fetch.ts#L5)

___

### fetchWrapper

▸ **fetchWrapper**(`url`, `method`, `data?`): `Promise`<`any`\>

#### Parameters

| Name | Type |
| :------ | :------ |
| `url` | `string` |
| `method` | `string` |
| `data?` | `any` |

#### Returns

`Promise`<`any`\>

#### Defined in

[src/ui/utils/fetch.ts:9](https://github.com/AdityaHegde/typescript-framework/blob/3b13972/src/ui/utils/fetch.ts#L9)

___

### getDisplayText

▸ **getDisplayText**(`model`): `string`

#### Parameters

| Name | Type |
| :------ | :------ |
| `model` | [`BaseType`](../classes/models.BaseType.md) |

#### Returns

`string`

#### Defined in

[src/ui/utils/getDisplayText.ts:3](https://github.com/AdityaHegde/typescript-framework/blob/3b13972/src/ui/utils/getDisplayText.ts#L3)

___

### getQueryString

▸ **getQueryString**(`query`, `filter?`, `parentKey?`, `level?`): `string`

#### Parameters

| Name | Type | Default value |
| :------ | :------ | :------ |
| `query` | `Dict`<`any`\> | `undefined` |
| `filter` | `QueryFilterType` | `undefined` |
| `parentKey` | `string` | `""` |
| `level` | `number` | `0` |

#### Returns

`string`

#### Defined in

[src/ui/utils/getQueryString.ts:3](https://github.com/AdityaHegde/typescript-framework/blob/3b13972/src/ui/utils/getQueryString.ts#L3)

___

### useRoutePath

▸ **useRoutePath**(`basePath?`): `string`

#### Parameters

| Name | Type | Default value |
| :------ | :------ | :------ |
| `basePath` | `string` | `""` |

#### Returns

`string`

#### Defined in

[src/ui/useRoutePath.ts:11](https://github.com/AdityaHegde/typescript-framework/blob/3b13972/src/ui/useRoutePath.ts#L11)
