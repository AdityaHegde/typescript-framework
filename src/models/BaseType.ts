import { ModelMetadata } from "./metadata/ModelMetadata";
import {ServerModelMetadata} from "./server-metadata/ServerModelMetadata";
import {ValidationMetadata} from "./validation-metadata/ValidationMetadata";

export class BaseType {
  public id: string;

  public type: string;

  public static metadata: ModelMetadata;
  public static serverMetadata: ServerModelMetadata;
  public static validationMetadata: ValidationMetadata;

  constructor(id: string) {
    this.type = (this.constructor as typeof BaseType).metadata.modelName;
    this.id = id;
  }
}
