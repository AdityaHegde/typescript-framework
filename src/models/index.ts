export {BaseType} from "./BaseType";
export {ModelList, MasterModelList, ModelNameMap, PolymorphicModelLookup, addModelsToList} from "./ModelList";
export {UserInvite} from "./UserInvite";
export {UserModel} from "./UserModel";

export {createMetadata} from "./metadata/createMetadata";
export {DisplayField} from "./metadata/DisplayField";
export {
  FieldType, FieldInputType, SimpleFieldType, SelectFieldType,
  ArrayFieldType, ModelSelectFieldType, DateFieldType,
  Field, TextField, NumberField, SwitchField, ArrayField,
  ModelMultiSelectField, ModelSelectField, MultiSelectField, SelectField,
  UpdatedAtDate, CreateAtDate,
} from "./metadata/Fields";
export {FilterType, LocalFilter, RemoteFilter, Filter} from "./metadata/Filters";
export {ModelMetadata, ModelFieldType, ModelArgsType} from "./metadata/ModelMetadata";
export {ParentKeyType, ParentKey} from "./metadata/ParentKey";
export {Readonly} from "./metadata/Readonly";
export {RelationType, HasOne, Relation, HasMany} from "./metadata/Relation";
export {Restricted} from "./metadata/Restricted";

export {createServerMetadata} from "./server-metadata/createServerMetadata";
export {RoleFieldType, DefaultRole, InviteeRole, RoleField, AdminRole} from "./server-metadata/RoleField";
export {ServerFieldType, ServerField, UniqueField, UniqueFieldGroup, IndexField} from "./server-metadata/ServerField";
export {ServerModelMetadata} from "./server-metadata/ServerModelMetadata";
export {UserField} from "./server-metadata/UserField";

export {createValidationMetadata} from "./validation-metadata/createValidationMetadata";
export {ValidationFieldType, RangeValidation, Validation, RegexValidation, EmailValidation, Required} from "./validation-metadata/Validation";
export {ValidationMetadata} from "./validation-metadata/ValidationMetadata";
