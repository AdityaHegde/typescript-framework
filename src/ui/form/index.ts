import {ArrayFormItem} from "./ArrayFormItem";
import {SwitchFormItem, FormItemType, FormItemWrapper, getFieldName, NumberFormItem, TextFormItem} from "./BasicFormItems";
import {ModelForm} from "./ModelForm";
import {ModelFormItems} from "./ModelFormItems";
import {ModelSelectFormItem} from "./ModelSelectFormItem";
import {SelectFormItem} from "./SelectFormItem";
import {useFormRef} from "./useFormRef";
import {getFormItem} from "./getFormItem";

export {
  ArrayFormItem,
  SwitchFormItem, FormItemType, FormItemWrapper, getFieldName, NumberFormItem, TextFormItem,
  ModelForm,
  getFormItem, ModelFormItems,
  ModelSelectFormItem,
  SelectFormItem,
  useFormRef,
}
