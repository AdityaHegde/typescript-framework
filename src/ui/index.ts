export {DataPage} from "./DataPage";
export {ModalWrapper} from "./ModalWrapper";
export {useRoutePath} from "./useRoutePath";

export {ComponentByModelMap} from "./utils/ComponentByModelMap";
export {fetchWrapper, fetchPost, fetchGet} from "./utils/fetch";
export {getDisplayText} from "./utils/getDisplayText";
export {getQueryString} from "./utils/getQueryString";
export {LightErrorBoundary} from "./utils/LightErrorBoundary";
export {SelectEnumItems} from "./utils/SelectEnumItems";
