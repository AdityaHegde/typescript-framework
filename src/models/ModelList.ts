import { BaseType } from "./BaseType";

/**
 * Full list of models
 */
export const ModelList: Array<typeof BaseType> = [];
/**
 * List of top level models
 */
export const MasterModelList: Array<typeof BaseType> = [];
export const ModelNameMap = new Map<string, typeof BaseType>();
export const PolymorphicModelLookup = new Map<string, Array<typeof BaseType>>();

function addModelToList(model: typeof BaseType) {
  if (!model.metadata.modelName) {
    return;
  }

  if (ModelNameMap.has(model.metadata.modelName)) {
    return;
  }

  ModelList.push(model);
  ModelNameMap.set(model.metadata.modelName, model);

  if (
    !model.metadata.polymorphic ||
    model.metadata.polymorphic === model.metadata.modelName
  ) {
    addModelToMasterList(model);
  }

  if (
    model.metadata.polymorphic &&
    model.metadata.polymorphic !== model.metadata.modelName
  ) {
    addModelToPolymorphicList(model);
  }
}

function addModelToMasterList(model: typeof BaseType) {
  MasterModelList.push(model);
}

function addModelToPolymorphicList(model: typeof BaseType) {
  if (!PolymorphicModelLookup.has(model.metadata.polymorphic)) {
    PolymorphicModelLookup.set(model.metadata.polymorphic, []);
  } else if (PolymorphicModelLookup.get(model.metadata.polymorphic).indexOf(model) !== -1) {
    // if the model was already added, dont add it again
    return;
  }

  PolymorphicModelLookup.get(model.metadata.polymorphic).push(model);
}

export function addModelsToList(models: Array<typeof BaseType>) {
  models.forEach((model) => {
    addModelToList(model);
  });
}
