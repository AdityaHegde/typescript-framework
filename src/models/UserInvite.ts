import {Validation} from "./validation-metadata/Validation";
import {Field, TextField} from "./metadata/Fields";
import {ServerModelMetadata} from "./server-metadata/ServerModelMetadata";
import {Readonly} from "./metadata/Readonly";
import {BaseType} from "./BaseType";
import {ServerField} from "./server-metadata/ServerField";
import {ModelMetadata} from "./metadata/ModelMetadata";

@ServerModelMetadata.Model({skipRoutes: true})
@ModelMetadata.Model()
export class UserInvite extends BaseType {
  @ServerField({unique: true, index: true})
  @Validation({required: true, email: true})
  @TextField({label: "Email ID"})
  public email: string;

  @Readonly()
  @TextField()
  public inviteToken: string;

  @Readonly()
  @Field({type: "date"})
  public expiry: Date;
}
