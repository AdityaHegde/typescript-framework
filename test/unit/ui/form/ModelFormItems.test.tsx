import React from "react";
import expect from "expect";
import renderer from "react-test-renderer";
import {SimpleModel} from "../../../test-classes/ui/SimpleModel";
import {AntdJestTestBase} from "../../../test-bases/AntdJestTestBase";
import {NestedModel} from "../../../test-classes/ui/NestedModel";
import {ExhaustivePropsModel} from "../../../test-classes/ui/ExhaustivePropsModel";
import {ModelForm} from "../../../../src/ui/form";
import { BaseType } from "../../../../src/models/BaseType";

@AntdJestTestBase.Suite
export class ModelFormItemsTest extends AntdJestTestBase {
  public modelFormData() {
    return {
      subData: [{
        title: "Simple models",
        args: [new SimpleModel("x"), SimpleModel],
      }, {
        title: "ExhaustivePropsModel models",
        args: [new ExhaustivePropsModel("x"), ExhaustivePropsModel],
      }, {
        title: "NestedModel models",
        args: [new NestedModel("x"), NestedModel],
      }],
    };
  }

  @AntdJestTestBase.Test("modelFormData")
  public modelFormTest(record: BaseType, model: typeof BaseType) {
    const tree = renderer.create(<ModelForm record={record} model={model} formRef={null} onFinish={() => {}} />);
    expect(tree).toMatchSnapshot();
  }
}
