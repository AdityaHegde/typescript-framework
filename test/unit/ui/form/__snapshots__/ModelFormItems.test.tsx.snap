// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelFormItemsTest modelFormTest ExhaustivePropsModel models 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-8 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="textField"
        title="textField"
      >
        textField
      </label>
    </div>
    <div
      className="ant-col ant-col-16 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="textField"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-8 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="numberField"
        title="Main Number Field"
      >
        Main Number Field
      </label>
    </div>
    <div
      className="ant-col ant-col-16 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-input-number"
            onBlur={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <div
              className="ant-input-number-handler-wrap"
            >
              <span
                aria-disabled={false}
                aria-label="Increase Value"
                className="ant-input-number-handler ant-input-number-handler-up"
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                role="button"
                unselectable="on"
              >
                <span
                  aria-label="up"
                  className="anticon anticon-up ant-input-number-handler-up-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                    />
                  </svg>
                </span>
              </span>
              <span
                aria-disabled={false}
                aria-label="Decrease Value"
                className="ant-input-number-handler ant-input-number-handler-down"
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                role="button"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  className="anticon anticon-down ant-input-number-handler-down-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              className="ant-input-number-input-wrap"
            >
              <input
                aria-valuenow="0"
                autoComplete="off"
                className="ant-input-number-input"
                id="numberField"
                onChange={[Function]}
                role="spinbutton"
                step={1}
                value="0"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-8 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="booleanField"
        title="booleanField"
      >
        booleanField
      </label>
    </div>
    <div
      className="ant-col ant-col-16 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="booleanField"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="checkbox"
            value={false}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-8 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="singleSelect"
        title="singleSelect"
      >
        singleSelect
      </label>
    </div>
    <div
      className="ant-col ant-col-16 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-select ant-select-single ant-select-show-arrow ant-select-show-search"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
          >
            <div
              className="ant-select-selector"
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="singleSelect_list_0"
                  aria-autocomplete="list"
                  aria-controls="singleSelect_list"
                  aria-haspopup="listbox"
                  aria-owns="singleSelect_list"
                  autoComplete="off"
                  className="ant-select-selection-search-input"
                  id="singleSelect"
                  onChange={[Function]}
                  onCompositionEnd={[Function]}
                  onCompositionStart={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onPaste={[Function]}
                  readOnly={false}
                  role="combobox"
                  style={
                    Object {
                      "opacity": null,
                    }
                  }
                  type="search"
                  unselectable={null}
                  value=""
                />
              </span>
              <span
                className="ant-select-selection-item"
                title="val1"
              >
                val1
              </span>
            </div>
            <span
              aria-hidden={true}
              className="ant-select-arrow"
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              <span
                aria-label="down"
                className="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-20 ant-col-sm-offset-4"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <button
              className="ant-btn ant-btn-dashed"
              onClick={[Function]}
              style={
                Object {
                  "width": "80%",
                }
              }
              type="button"
            >
              <span
                aria-label="plus"
                className="anticon anticon-plus"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="plus"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <defs>
                    <style />
                  </defs>
                  <path
                    d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                  />
                  <path
                    d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                  />
                </svg>
              </span>
              <span>
                 Add arrayField
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`ModelFormItemsTest modelFormTest NestedModel models 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-8 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="textField"
        title="textField"
      >
        textField
      </label>
    </div>
    <div
      className="ant-col ant-col-16 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="textField"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-8 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="simpleModelId"
        title="simpleModelId"
      >
        simpleModelId
      </label>
    </div>
    <div
      className="ant-col ant-col-16 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-select ant-select-single ant-select-show-arrow ant-select-show-search"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
          >
            <div
              className="ant-select-selector"
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="simpleModelId_list_0"
                  aria-autocomplete="list"
                  aria-controls="simpleModelId_list"
                  aria-haspopup="listbox"
                  aria-owns="simpleModelId_list"
                  autoComplete="off"
                  className="ant-select-selection-search-input"
                  id="simpleModelId"
                  onChange={[Function]}
                  onCompositionEnd={[Function]}
                  onCompositionStart={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onPaste={[Function]}
                  readOnly={false}
                  role="combobox"
                  style={
                    Object {
                      "opacity": null,
                    }
                  }
                  type="search"
                  unselectable={null}
                  value=""
                />
              </span>
              <span
                className="ant-select-selection-placeholder"
              />
            </div>
            <span
              aria-hidden={true}
              className="ant-select-arrow"
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              <span
                aria-label="down"
                className="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`ModelFormItemsTest modelFormTest Simple models 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-8 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="textField"
        title="textField"
      >
        textField
      </label>
    </div>
    <div
      className="ant-col ant-col-16 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="textField"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-8 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="singleSelect"
        title="singleSelect"
      >
        singleSelect
      </label>
    </div>
    <div
      className="ant-col ant-col-16 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-select ant-select-single ant-select-show-arrow ant-select-show-search"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
          >
            <div
              className="ant-select-selector"
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="singleSelect_list_0"
                  aria-autocomplete="list"
                  aria-controls="singleSelect_list"
                  aria-haspopup="listbox"
                  aria-owns="singleSelect_list"
                  autoComplete="off"
                  className="ant-select-selection-search-input"
                  id="singleSelect"
                  onChange={[Function]}
                  onCompositionEnd={[Function]}
                  onCompositionStart={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onPaste={[Function]}
                  readOnly={false}
                  role="combobox"
                  style={
                    Object {
                      "opacity": null,
                    }
                  }
                  type="search"
                  unselectable={null}
                  value=""
                />
              </span>
              <span
                className="ant-select-selection-placeholder"
              />
            </div>
            <span
              aria-hidden={true}
              className="ant-select-arrow"
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              <span
                aria-label="down"
                className="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-8 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="nestedModelId"
        title="nestedModelId"
      >
        nestedModelId
      </label>
    </div>
    <div
      className="ant-col ant-col-16 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-select ant-select-single ant-select-show-arrow ant-select-show-search"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
          >
            <div
              className="ant-select-selector"
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="nestedModelId_list_0"
                  aria-autocomplete="list"
                  aria-controls="nestedModelId_list"
                  aria-haspopup="listbox"
                  aria-owns="nestedModelId_list"
                  autoComplete="off"
                  className="ant-select-selection-search-input"
                  id="nestedModelId"
                  onChange={[Function]}
                  onCompositionEnd={[Function]}
                  onCompositionStart={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onPaste={[Function]}
                  readOnly={false}
                  role="combobox"
                  style={
                    Object {
                      "opacity": null,
                    }
                  }
                  type="search"
                  unselectable={null}
                  value=""
                />
              </span>
              <span
                className="ant-select-selection-placeholder"
              />
            </div>
            <span
              aria-hidden={true}
              className="ant-select-arrow"
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              <span
                aria-label="down"
                className="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
